src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceContext.java

```diff
@@ -68,6 +73,12 @@
     @NonNull
     private Set<ChangeRequestCheckoutStrategy> forkPRStrategies = EnumSet.noneOf(ChangeRequestCheckoutStrategy.class);
     /**
+     * Strategies used to notify Github of build status.
+     *
+     * @since TODO
+     */
+    private List<AbstractGitHubNotificationStrategy> notificationStrategies = new ArrayList<>();
+    /**
      * {@code true} if notifications should be disabled in this context.
      */
     private boolean notificationsDisabled;
```

Comment:
I think we can elide this field by using `notificationStrategies.isEmpty()` as the read and have the write of true restore the list of strategies to the default list (if the list was empty, no-op if the list non-empty)

Reply:
Do you mean a write of false?

Reply:
```
/* names changed because I'm in a browser and cannot look up easily */
in withNotificationsDisabled(boolean newValue) {
  if (newValue) {
    if (notificationStrategies.isEmpty()) {
      notificationStrategies.add(new DefaultGitHubNotificationStrategy());
    }
  } else {
    if (!notificationStrategies.isEmpty()) {
      notificationStrategies.clear();
    }
  }
}
```

Reply:
The idea is that we want to reduce the effective state that we need to maintain. Previously disabling notifications was a simple boolean. But now we have a list of strategies and an empty list is exactly equivalent to disabled. We probably could deprecate the `withNotificationsDisabled` mutator as we'd rather people just set an empty list if they want to disable... but it seems a reasonable utility method. Better instead to document that turning notification on when they are off will reset the notifications to the default